public/app/features/explore/QueryLibrary/QueryTemplateForm.tsx

```diff
@@ -82,7 +88,13 @@ export const QueryTemplateForm = ({ onCancel, onSave, queryToAdd, templateData }
   };
 
   const handleEditQueryTemplate = async (editQueryTemplateCommand: EditQueryTemplateCommand) => {
-    return editQueryTemplate(editQueryTemplateCommand)
+    return editQueryTemplate({
+      namespace: getK8sNamespace(),
+      name: editQueryTemplateCommand.uid,
+      ioK8SApimachineryPkgApisMetaV1Patch: {
```

Comment:
same here... ugly name, but it's ugly in the api spec. can we fix this?

Reply:
It's written in the spec as `io.k8s.apimachinery.pkg.apis.meta.v1.Patch`, so we'd probably remove the `io.k8s.apimachinery.pkg.apis.meta.v1` part when generating it (this will also remove it from the documentation, e.g., Swagger, which might not be what we want). The more I look at stuff like this the more I think that we might need some post-processing script here.

Reply:
you mean when generating the rtk query client, or when generating the spec itself? 🤔 idk how the backend works, but i don't think having `io.k8s.apimachinery.pkg.apis.meta.v1` as a field on an api is particularly nice tbh 😅

Reply:
ahhhh nevermind i think i understand... the problem is in the client generation. the spec just says the body has to be of this schema (`ioK8SApimachineryPkgApisMetaV1Patch`), so for a user of the api they just do something like:
```tsx
fetch({
  url,
  method: 'PATCH',
  body: <bodyWithTypeioK8SApimachinery...>
})
```

but when creating the api client, the code generation uses that schema name to generate the field name 😬 that is unfortunate 😭

Reply:
The issue here is that this is used on only on frontend but also inside k8s, and in k8s context all these verbose names apparently make sense 🤷

Reply:
> ahhhh nevermind i think i understand... 

Yeah, the name doesn't really matter, we can even rename it to `body`, e.g., https://github.com/grafana/grafana/pull/98891/files#diff-b486d2c9aad66e187e00e161ed7d846f37fc8cf2e8a93bbca0525f3df394ac50R34

Reply:
i wonder why `rtk-query-codegen-api` does it in the first place...? 🤔 

there's this [open issue](https://github.com/reduxjs/redux-toolkit/issues/2961) about basically the same thing. it does make using the hooks particularly unreadable 😬

Reply:
It's defined like this at the bottom of the spec: 
```
      "io.k8s.apimachinery.pkg.apis.meta.v1.Patch": {
        "description": "Patch is provided to give a concrete name and type to the Kubernetes PATCH request body.",
        "type": "object"
      },
```

I guess the codegen just uses that to create the type name.

Reply:
yeah it's weird... the code to generate the name is [here](https://github.com/reduxjs/redux-toolkit/blob/master/packages/rtk-query-codegen-openapi/src/generate.ts#L338). if there's a naming conflict it defaults back to `body` 🤔 in my mind, it should start with `body` and only fall back to the more verbose names if there's a conflict